api-presigned.go

```diff
@@ -122,21 +122,41 @@ func (c Client) PresignedPostPolicy(p *PostPolicy) (u *url.URL, formData map[str
 		return nil, nil, err
 	}
 
+	// Get credentials from the configured credentials provider.
+	value, err := c.credsProvider.Get()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	signerType := c.signature
+	// Check if the signature type is default, it means caller didn't
+	// ask for anything specific signature style.. Check if the
+	// credentials set returned the type of signature that should
+	// be used, if not proceed to use default.
+	if signerType == SignatureDefault && value.SignatureType != "" {
```

Comment:
I think that the signature type that comes from config, if it exists, should override the one specified in the application code. That way, we can give users more power to configure.